### PR TITLE
Scripts - Add Forced shooting function

### PR DESCRIPTION
**Pull request by TyroneMF**
- Function to force an AI to fire at a target until either are dead or time has elapsed.

### CODE REVIEW

File: Scripts/fn_forceShooting.sqf [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+ * Firemode types Single (0) and Full auto (1)
+ * Set firing timer to 0.1 for full auto
+ * Designed ONLY for infantry, vehicle results not guaranteed.
+ *
+ * Arguments:
+ * 0: AI Shooter <OBJECT>
+ * 1: Target <OBJECT>
+ * 2: Firemode <NUMBER> (default: 0)
+ * 3: Timer <NUMBER> (default: 10)
+ * 4: Time between shots <NUMBER> (default: 1)
```
Comment (jonpas) [inline Scripts/fn_forceShooting.sqf:14]: Is `0.1` the lowest it should ever be set? What happens if you set it to `0`?

File: Scripts/fn_forceShooting.sqf [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+private _ehIndex = _shooter addEventHandler ["Reloaded", {
+    params ["_unit", "", "", "_newMagazine"];
+    private _magazine = _newMagazine select 0;
+    _unit addMagazine _magazine;
+}];
+
+// Wait for unit to actually target.
+[{
+    private _array = _this select 0;
+    private _firingTimer = _array select 4;
```
Comment (jonpas) [inline Scripts/fn_forceShooting.sqf:47]:
```suggestion
    private _firingTimer = (_this select 0) select 4;
```

File: Scripts/fn_forceShooting.sqf [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+
+    [{
+        params ["_args", "_handle"];
+        _args params ["_functionArgs", "_additionalArgs"];
+        _functionArgs params ["_shooter", "_target", "", "_timer", "_firingTimer"];
```
Comment (jonpas) [inline Scripts/fn_forceShooting.sqf:52]:
```suggestion
        _functionArgs params ["_shooter", "_target", "", "_timer"];
```
Unused.

File: Scripts/fn_forceShooting.sqf [hunk 4]
```diff
@@ -0,0 +1,65 @@
+#include "..\script_component.hpp"
+/*
+ * Author: Jonpas, Rory, Tyrone
+ * Orders a unit to target an object and begin firing until either the shooter or target is dead or the timer elapses.
+ * Firemode types Single (0) and Full auto (1)
+ * Set firing timer to 0.1 for full auto
+ * Designed ONLY for infantry, vehicle results not guaranteed.
```
Comment (jonpas) [inline Scripts/fn_forceShooting.sqf:7]:
```suggestion
 * Designed ONLY for infantry, using with vehicles will result in undefined behaviour.
```

**Review by jonpas — CHANGES_REQUESTED** [review]

File: Scripts/fn_forceShooting.sqf (hunk 1 above)
Comment (TyroneMF) [reply Scripts/fn_forceShooting.sqf:14]: Nothing really, just figured for no difference it's probably better to not run it on every frame.

Comment (jonpas) [reply Scripts/fn_forceShooting.sqf:14]: You can limit it then.
```
_firingTimer = _firingTimer max 0.1
```

**Review by jonpas — APPROVED** [review]

**Pull request merged**